docs/docs/dev_docs/testing/cheat_codes.md

```diff
@@ -0,0 +1,449 @@
+public async dumpChainState(fileName: string): Promise<void> 
+
+// Loads the L1 chain state from a file. You may use `dumpChainState()` to save the state of the L1 chain to a file and later load it. 
+public async loadChainState(fileName: string): Promise<void> 
+
+// Load the value at a storage slot of a contract address on L1
+public async load(contract: EthAddress, slot: bigint): Promise<bigint> 
+
+// Set the value at a storage slot of a contract address on L1 (e.g. modify a storage variable on your portal contract or even the rollup contract). 
+public async store(contract: EthAddress, slot: bigint, value: bigint): Promise<void> 
+
+// Computes the slot value for a given map and key on L1. A convenient wrapper to find the appropriate storage slot to load or overwrite the state.
+public keccak256(baseSlot: bigint, key: bigint): bigint
+
+// Send transactions on L1 impersonating an externally owned account or contract.
+public async startPrank(who: EthAddress): Promise<void> 
+
+// Stop impersonating an account on L1 that you are currently impersonating.
+public async stopPrank(who: EthAddress): Promise<void> 
+
+// Set the bytecode for a L1 contract
+public async etch(contract: EthAddress, bytecode: `0x${string}`): Promise<void> 
+
+// Get the bytecode for a L1 contract
+public async getBytecode(contract: EthAddress): Promise<`0x${string}`> 
+```
+
+### blockNumber
+
+#### Function Signature 
+```js
+public async blockNumber(): Promise<number>
+```
+
+#### Description
+Fetches the current L1 block number
+
+#### Example
+```js
+const blockNumber = await cc.l1.blockNumber()
+```
+
+### chainId
+
+#### Function Signature 
+```js
+public async chainId(): Promise<number>  
+```
+
+#### Description
+Fetches the L1 chain ID
+
+#### Example 
+```js
+const chainId = await cc.l1.chainId()
+```
+
+### timestamp
+
+#### Function Signature
+```js
+public async timestamp(): Promise<number> 
+```
+
+#### Description 
+Fetches the current L1 timestamp
+
+#### Example
+```js 
+const timestamp = await cc.l1.timestamp()
+```
+
+### mine
+
+#### Function Signature
+```js
+public async mine(numberOfBlocks = 1): Promise<void> 
+```
+
+#### Description
+Mines the specified number of blocks on L1 (default 1)
+
+#### Example
+```js
+const blockNum = await cc.l1.blockNumber();
+await cc.l1.mine(10) // mines 10 blocks
+const newBlockNum = await cc.l1.blockNumber(); // = blockNum + 10.
+```
+
+### setNextBlockTimestamp
+
+#### Function Signature  
+```js
+public async setNextBlockTimestamp(timestamp: number): Promise<void>
+```
+
+#### Description
+Sets the timestamp (unix format in seconds) for the next mined block on L1  
+Remember that timestamp can only be set in the future and not in the past.
+
+#### Example
+```js
+await cc.l1.setNextBlockTimestamp(1692183270) // Set next block timestamp to 16 Aug 2023 10:54:30 GMT
```

Comment:
The comment here will most often go out of the field of view. Probably better to have on the line above etc.